source/NVDAObjects/UIA/__init__.py

```diff
@@ -18,36 +19,98 @@
-			except COMError:
-				fontSizeValue=UIAHandler.handler.reservedNotSupportedValue
-			if fontSizeValue!=UIAHandler.handler.reservedNotSupportedValue:
-				formatField['font-size']="%g pt"%float(fontSizeValue)
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_FontSizeAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if isinstance(val,numbers.Number):
+				formatField['font-size']="%g pt"%float(val)
+		if formatConfig["reportFontAttributes"]:
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_FontWeightAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if isinstance(val,int):
+				formatField['bold']=(val>=700)
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_IsItalicAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if not UIAHandler.handler.clientObject.checkNotSupported(val):
+				formatField['italic']=val
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_UnderlineStyleAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if not UIAHandler.handler.clientObject.checkNotSupported(val):
+				formatField['underline']=bool(val)
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_StrikethroughStyleAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if not UIAHandler.handler.clientObject.checkNotSupported(val):
+				formatField['strikethrough']=bool(val)
+			textPosition=None
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_IsSuperscriptAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if not UIAHandler.handler.clientObject.checkNotSupported(val) and val:
+				textPosition='super'
+			else:
+				val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_IsSubscriptAttributeId,ignoreMixedValues=ignoreMixedValues)
+				if not UIAHandler.handler.clientObject.checkNotSupported(val) and val:
+					textPosition="sub"
+				else:
+					textPosition="baseline"
+			if textPosition:
+				formatField['text-position']=textPosition
+		if formatConfig["reportAlignment"]:
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_HorizontalTextAlignmentAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if val==UIAHandler.HorizontalTextAlignment_Left:
+				val="left"
+			elif val==UIAHandler.HorizontalTextAlignment_Centered:
+				val="center"
+			elif val==UIAHandler.HorizontalTextAlignment_Right:
+				val="right"
+			elif val==UIAHandler.HorizontalTextAlignment_Justified:
+				val="justify"
+			else:
+				val=None
+			if val:
+				formatField['text-align']=val
+		if formatConfig["reportColor"]:
+			import colors
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_BackgroundColorAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if isinstance(val,int):
+				formatField['background-color']=colors.RGB.fromCOLORREF(val)
+			val=getUIATextAttributeValueFromRange(range,UIAHandler.UIA_ForegroundColorAttributeId,ignoreMixedValues=ignoreMixedValues)
+			if isinstance(val,int):
+				formatField['color']=colors.RGB.fromCOLORREF(val)
+		if formatConfig['reportLinks']:

```

Comment:
I thought links worked in Edge before this; e.g. in NVDA 2016.3. Are links now handled differently in Edge or is this for Mail?